server/public/model/license.go

```diff
@@ -459,9 +473,28 @@ func (lr *LicenseRecord) PreSave() {
 	lr.CreateAt = GetMillis()
 }
 
-func MinimumProfessionalProvidedLicense(license *License) *AppError {
-	if license == nil || (license.SkuShortName != LicenseShortSkuProfessional && license.SkuShortName != LicenseShortSkuEnterprise) {
-		return NewAppError("", NoTranslation, nil, "license is neither professional nor enterprise", http.StatusNotImplemented)
+// MinimumProfessionalLicense returns true if the provided license is at least a professional license.
+// Higher tier licenses also satisfy the condition.
+func MinimumProfessionalLicense(license *License) *AppError {
+	if license == nil || LicenseToLicenseTier[license.SkuShortName] < ProfessionalTier {
+		return NewAppError("", NoTranslation, nil, "license is neither professional nor enterprise nor premium", http.StatusNotImplemented)
+	}
+	return nil
+}
+
+// MinimumEnterpriseLicense returns true if the provided license is at least a enterprise license.
+// Higher tier licenses also satisfy the condition.
+func MinimumEnterpriseLicense(license *License) *AppError {
+	if license == nil || LicenseToLicenseTier[license.SkuShortName] < EnterpriseTier {
+		return NewAppError("", NoTranslation, nil, "license is neither enterprise nor premium", http.StatusNotImplemented)
+	}
+	return nil
+}
+
+// MinimumPremiumLicense returns true if the provided license is at least a premium license.
+func MinimumPremiumLicense(license *License) *AppError {
+	if license == nil || LicenseToLicenseTier[license.SkuShortName] < PremiumTier {
+		return NewAppError("", NoTranslation, nil, "license is not premium", http.StatusNotImplemented)
 	}
 	return nil
 }
```

Comment:
Would it be simpler to return a boolean from these functions? It seems that more frequently checks on these return values are whether or not they're `nil`, so a boolean would be cleaner. I thought we were trying to move away from `AppError`, so places that require regular `error` type will have to do a bunch of wrapping, etc. Using a simple boolean, you can let the caller decide what they should do with the error.

Reply:
Done.